osa.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python
```

Comment:
Nitpick: Nice modularization (having your command functions in `main.py` then using them in here) but is it really necessary?